compose/run.sh

```diff
@@ -17,35 +36,62 @@ description:
     file first, then rebuild dgraph and bring up the config.
 EOF
     exit 0
-fi
+}
 
-function Info {
-    echo -e "INFO: $*"
+check_environment() {
+  command -v make > /dev/null || \
+    { echo "ERROR: 'make' command not not found" 1>&2; exit 1; }
+  command -v go > /dev/null || \
+    { echo "ERROR: 'go' command not not found" 1>&2; exit 1; }
+  command -v docker-compose > /dev/null || \
+    { echo "ERROR: 'docker-compose' command not not found" 1>&2; exit 1; }
+  ## GOPATH required for locally built docker images
+  [[ -z "${GOPATH}" ]] && \
+    { echo "ERROR: The env var of 'GOPATH' was not defined. Exiting" 1>&2; exit 1; }
 }
 
-#
-# MAIN
-#
+build_compose_tool() {
+  ## Always make compose if it doesn't exist
+  make compose
 
-set -e
+  ## Create compose file if it does not exist or compose parameters passed
+  if [[ $# -gt 0 ]] || ! [[ -f $COMPOSE_FILE ]]; then
+      Info "creating compose file ..."
+      ./compose "$@"
+  fi
 
-if [[ $# -gt 0 ]]; then
-    Info "creating compose file ..."
-    go build compose.go
-    ./compose "$@"
-fi
+  if [[ ! -e $COMPOSE_FILE ]]; then
+      echo >&2 "$ME: no '$COMPOSE_FILE' found"
+      exit 1
+  fi
+}
 
-if [[ ! -e $COMPOSE_FILE ]]; then
-    echo >&2 "$ME: no $COMPOSE_FILE found"
+build_dgraph_docker_image() {
+  ## linux binary required for docker image
+  export GOOS=linux
+  Info "rebuilding dgraph ..."
+  ( cd $DGRAPH_ROOT/dgraph && make install )
+}
+
+launch_environment() {
+  # Detect if $GOPATH/bin/$GOOS_$GOARCH path
+  if [[ -f $GOPATH/bin/linux_amd64/dgraph ]]; then
+    Info "Found '$GOPATH/bin/linux_amd64/dgraph'.  Updating $COMPOSE_FILE."
+    sed -i 's/\$GOPATH\/bin$/\$GOPATH\/bin\/linux_amd64/' $COMPOSE_FILE
```

Comment:
This should go into the compose tool, not run.sh. This script shouldn't modify the docker-compose.yml.

Reply:
Follow-up PR.